lib/rdl/types/type.rb

```diff
@@ -5,6 +5,7 @@ module RDL::Type
   class TypeError < StandardError; end
 
   class Type
+    attr_accessor :solution
```

Comment:
@jeffrey-s-foster I put this here so that the standard behavior for every type was to have a solution field that can be read to and written from.

Reply:
Actually, I don't know that this should always be a field. Things like `NominalType` don't need a file. So maybe the invariant is  just that  there is always a `solution` method, and for some things it returns a field?

Reply:
Ok, that makes sense. I'd kind of inverted this in that `MethodType` has its own `solution` method that overrides it. Should the default `solution` method just return `self`?